pkg/git/repository.go

```diff
@@ -121,3 +122,38 @@ func (r *Repository) GetLatestCommit() (*object.Commit, error) {
 	}
 	return commit, nil
 }
+
+func (r *Repository) CommitDeployment(DeploymentFileRepositoryPath, KubernetesDeploymentFilePath string) error {
+	// get worktree
+	w, err := r.Worktree()
+	if err != nil {
+		return ErrGeneratingWorkTree{}
+	}
+
+	fileRelToRepo, err := filepath.Rel(DeploymentFileRepositoryPath,KubernetesDeploymentFilePath )
+	// git add deploymentFile
+	_, err = w.Add(fileRelToRepo)
+	if err != nil {
+		return ErrGitAdd{}
+	}
+
+	_, err = w.Commit("Kubernite deployment", &goGit.CommitOptions{
+		Author: &object.Signature{
+			Name:  "Kubernite",
+			Email: "-",
+			When:  time.Now(),
+		},
+	})
+
+	if err != nil {
```

Comment:
consider placing all things calls with a, err := func()... inside if statement if a is being ignored